pkg/controller/acmechallenges/finalizer.go

```diff
@@ -30,18 +30,12 @@ import (
 // allowing the garbage collector to remove the challenge.
 
 // finalizerRequired returns true if the finalizer is not found on the challenge.
-//
-// API transition
-// We currently only add cmacme.ACMELegacyFinalizer, but a future version will add
-// cmacme.ACMEDomainQualifiedFinalizer.
-// A finalizer only needs to be added if neither is present.
 func finalizerRequired(ch *cmacme.Challenge) bool {
-	finalizers := sets.NewString(ch.Finalizers...)
-	return !finalizers.Has(cmacme.ACMELegacyFinalizer) &&
-		!finalizers.Has(cmacme.ACMEDomainQualifiedFinalizer)
+	return !sets.NewString(ch.Finalizers...).Has(cmacme.ACMEDomainQualifiedFinalizer)
 }
 
 func otherFinalizerPresent(ch *cmacme.Challenge) bool {
-	return ch.Finalizers[0] != cmacme.ACMELegacyFinalizer &&
-		ch.Finalizers[0] != cmacme.ACMEDomainQualifiedFinalizer
+	finalizers := sets.NewString(ch.Finalizers...).
+		Delete(cmacme.ACMEDomainQualifiedFinalizer, cmacme.ACMELegacyFinalizer)
+	return len(finalizers) > 0
```

Comment:
- This change addresses https://github.com/cert-manager/cert-manager/pull/7735/#discussion_r2075871787

Reply:
This seems address a known issue where we assume that the "other finalizer" is the first.
I think this overlaps with a similar change in https://github.com/cert-manager/cert-manager/pull/7286 (originally https://github.com/cert-manager/cert-manager/pull/5126). 
Seems like an improvement though.